src/components/producerItems/actionTypes.js

```diff
@@ -0,0 +1,33 @@
+import { NAME } from './constants';
+
+/**
+ * Fired by the {@link actions.requestProductList requestProductList}
+ * action creator.
+ *
+ * @type {String}
+ */
+export const REQUEST_PRODUCT_LIST = `${NAME}/REQUEST_PRODUCT_LIST`;
+
+/**
+ * Fired by the {@link actions.requestProductUpdate requestProductUpdate}
+ * action creator.
+ *
+ * @type {String}
+ */
+export const REQUEST_PRODUCT_UPDATE = `${NAME}/REQUEST_PRODUCT_UPDATE`;
+
+/**
+ * Fired by the {@link actions.updateProductDetails updateProductDetails}
+ * action creator.
+ *
+ * @type {String}
+ */
+export const UPDATE_PRODUCT_DETAILS = `${NAME}/UPDATE_PRODUCT_DETAILS`;
+
+/**
+ * Fired by the {@link actions.updateProductList updateProductList}
+ * action creator.
+ *
+ * @type {String}
+ */
+export const UPDATE_PRODUCT_LIST = `${NAME}/UPDATE_PRODUCT_LIST`;
```

Comment:
Why exactly are these action creators written this way? Is it a path?

Reply:
These are `action types` and the reason that i am prefixing it with the name of the component is so that when we look at the actions through the redux dev tool we can easily see where the action is coming from. As you app grows the likeliness that you will get duplicate action types will increase, this is a simple fix to address that.